org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/FormatterHandler.java

```diff
@@ -85,7 +86,13 @@ public FormatterHandler(PreferenceManager preferenceManager) {
 				return Collections.<org.eclipse.lsp4j.TextEdit>emptyList();
 			}
 			String sourceToFormat = document.get();
-			TextEdit format = formatter.format(CodeFormatter.K_COMPILATION_UNIT, sourceToFormat, region.getOffset(), region.getLength(), 0, lineDelimiter);
+			int kind;
+			if (cu.getResource() != null && cu.getResource().getName().equals(IModule.MODULE_INFO_JAVA)) {
+				kind = CodeFormatter.K_MODULE_INFO;
+			} else {
+				kind = CodeFormatter.K_COMPILATION_UNIT | (preferenceManager.getPreferences().isJavaFormatComments() ? CodeFormatter.F_INCLUDE_COMMENTS : 0);
+			}

```

Comment:
```java
if (cu.getResource() != null && cu.getResource().getName().equals(IModule.MODULE_INFO_JAVA)) {
	kind = CodeFormatter.K_MODULE_INFO;
} else {
	kind = CodeFormatter.K_COMPILATION_UNIT;
}
kind |= (preferenceManager.getPreferences().isJavaFormatComments() ? CodeFormatter.F_INCLUDE_COMMENTS : 0);
```

Reply:
I have copied the code from https://github.com/eclipse/eclipse.jdt.core/blob/master/org.eclipse.jdt.core/formatter/org/eclipse/jdt/core/formatter/CodeFormatterApplication.java#L225

Reply:
no, you missed the parentheses around the ternary operation

Reply:
Fixed it.